app/channels/collect_submissions_channel.rb

```diff
@@ -0,0 +1,14 @@
+class CollectSubmissionsChannel < ApplicationCable::Channel
+  def subscribed
+    unless allowed_to?(:collect_submissions?, with: SubmissionPolicy, context: { real_user: current_user,
+                                                                                 role: current_user,
```

Comment:
Roles are separate from users: they act as a join model between users and courses. So it's quite strange here that `current_user` (a `User`) is being passed as `role`. Are you confident the policy is working as intended here? As a comment, some of the tests below in `collect_submissions_channel_spec.rb` are stubbing `current_user` as an instructor/student (i.e., as a `Role`) rather than a `User`, so they might not be checking what you intend.

The tricky part is that you need a `course_id` in order to compute the associated role for this user. I think you can pass it as a parameter when creating the subscription from the front-end?

Reply:
(and to be clear, when you move this code into the React component at I mentioned below, you should have access to the props.course_id)